filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala

```diff
@@ -157,18 +168,76 @@ class GcsPathBuilder(apiStorage: com.google.api.services.storage.Storage,
 
 case class GcsPath private[gcs](nioPath: NioPath,
                                 apiStorage: com.google.api.services.storage.Storage,
-                                cloudStorage: com.google.cloud.storage.Storage
-                               ) extends Path {
+                                cloudStorage: com.google.cloud.storage.Storage,
+                                projectId: String) extends Path {
   lazy val blob = BlobId.of(cloudStoragePath.bucket, cloudStoragePath.toRealPath().toString)
 
-  override protected def newPath(nioPath: NioPath): GcsPath = GcsPath(nioPath, apiStorage, cloudStorage)
+  override protected def newPath(nioPath: NioPath): GcsPath = GcsPath(nioPath, apiStorage, cloudStorage, projectId)
 
   override def pathAsString: String = {
     val host = cloudStoragePath.bucket().stripSuffix("/")
     val path = cloudStoragePath.toString.stripPrefix("/")
     s"${CloudStorageFileSystem.URI_SCHEME}://$host/$path"
   }
 
+  /***
+    * This method needs to be overridden to make it work with requester pays. We need to go around Nio
+    * as currently it doesn't support to set the billing project id. Google Cloud Storage already has
+    * code in place to set the billing project (inside HttpStorageRpc) but Nio does not pass it even though
+    * it's available. In future when it is supported, remove this method and wire billing project into code
+    * wherever necessary
+    */
+  override def mediaInputStream: InputStream = {
+    Try{
+      apiStorage.objects().get(blob.getBucket, blob.getName).setUserProject(projectId).executeMediaAsInputStream()
+    } match {
+      case Success(inputStream) => inputStream
+      case Failure(e) => throw new IOException(s"Creating an input stream failed during executeMediaAsInputStream() because of ${e.getMessage}")
```

Comment:
ToL: The mention of `executeMediaAsInputStream` seems a bit too specific. This message will be surfaced to the user and I don't think this would be useful to them. I'd also add the path for which we failed to open a stream in the message :)

Reply:
You can also add the whole exception as a cause to the IoException instead of concatenating the message, like 
`throw new IOException(s"Could not open an input stream to $pathAsString", e)`

Reply:
Yes, it makes sense. I will make it less specific.